src/controllers/user.controller.ts

```diff
@@ -202,3 +297,116 @@ export const getFriends = async (req: Request, res: Response) => {
     return res.status(500).send({ message: error?.message });
   }
 };
+
+/**
+ * Sends a request to add a friend for the user with the specified userId.
+ * @param {Request} req - Express request object containing parameters: userId and friendId.
+ * @param {Response} res - Express response object used to send the response back to the client.
+ * @returns {Promise} Returns status code 200. Otherwise, returns an error status with an appropriate message.
+ */
+export const requestFriend = async (req: Request, res: Response) => {
+  const userId = req.params?.userId || '';
+  const friendId = req.query?.friendId?.toString() || '';
+
+  if (!ObjectId.isValid(userId)) {
+    return res.status(400).send({ message: 'Invalid user ID!' });
+  }
+
+  if (!ObjectId.isValid(friendId)) {
+    return res.status(400).send({ message: 'Invalid friend ID!' });
+  }
+
+  try {
+    const targetUser = await User.findByIdAndUpdate(friendId, {
+      $push: { friendRequests: userId },
+    });
+
+    if (targetUser === null) {
+      return res.status(400).send({ message: 'User does not exist!' });
+    }
+
+    return res
+      .status(200)
+      .send({ message: 'Successfully sent friend request!' });
+  } catch (error: any) {
+    return res.status(500).send({ message: error?.message });
+  }
+};
+
+/**
+ * Accepts friend request and adds the new friend to user's friends list while removing the request from
+ * friendRequests list.
+ * @param {Request} req - Express Request object
+ * @param {Response} res - Express Response object
+ * @returns {Promise} Returns success message on successful addition of friend, otherwise returns an error message
+ * with corresponding status code.
+ */
+export const acceptFriendRequest = async (req: Request, res: Response) => {
+  const userId = req.params?.userId || '';
+  const friendId = req.query?.friendId?.toString() || '';
+
+  if (!ObjectId.isValid(userId)) {
+    return res.status(400).send({ message: 'Invalid user ID!' });
+  }
+
+  if (!ObjectId.isValid(friendId)) {
+    return res.status(400).send({ message: 'Invalid friend ID!' });
+  }
+
+  try {
+    const targetUser = await User.findByIdAndUpdate(userId, {
+      $push: { friends: friendId },
+      $pull: { friendRequests: friendId },
+    });
+
+    const targetFriendUser = await User.findByIdAndUpdate(friendId, {
+      $push: { friends: userId },
+    });
+
+    if (targetUser === null || targetFriendUser === null) {
+      return res.status(400).send({ message: 'User does not exist!' });
+    }
```

Comment:
I would consider splitting this into two conditions so we can send back more specific error messages.